bin/git-archive-file

```diff
@@ -20,6 +20,8 @@ else
 	if [ "$BRANCH" = "master" ]; then
 		FILENAME=$ARCHIVE_NAME.$VERSION.zip
 	else
+	  # govern against invalid path characters
+	  BRANCH=$(echo "$BRANCH" | sed -r 's/[\/\\\:]+/_/g')
```

Comment:
As `g` option is specific, is it necessary to use `+` in the regex?
If `+` can be omitted, `-r` can be omitted too.

Reply:
It does not hurt, and covers all future scenarios.

Reply:
@DarekDan
Less code, less bugs.
For example, bsd `sed` doesn't have the `-r` option. If the extended regex is required, we have to write more code to work around.